overlord/snapstate/refresh_test.go

```diff
@@ -153,19 +126,20 @@ func (s *refreshSuite) TestHardNothingRunningRefreshCheck(c *C) {
 	s.info.Apps["daemon"].RefreshMode = ""
 
 	// Applications are also blocking hard refresh check.
-	writePids(c, s.daemonPath, []int{})
-	writePids(c, s.appPath, []int{101})
+	s.pids = map[string][]int{
+		"snap.pkg.app": {101},
+	}
 	err = snapstate.HardNothingRunningRefreshCheck(s.info)
 	c.Assert(err, NotNil)
-	c.Check(err.Error(), Equals, `snap "foo" has running apps (app)`)
+	c.Check(err.Error(), Equals, `snap "pkg" has running apps (app)`)
 	c.Check(err.(*snapstate.BusySnapError).Pids(), DeepEquals, []int{101})
```

Comment:
maybe a `FitsTypeOf` assertion above this, as if the error isn't of that type this will panic which doesn't produce super readable test results on failure

Reply:
I wasn't familiar with this checker before. Done!